lib/code_corps/github/webhook/handler.ex

```diff
@@ -11,46 +11,59 @@ defmodule CodeCorps.GitHub.Webhook.Handler do
     GitHub.Event.IssueComment,
     GitHub.Event.Issues,
     GitHub.Event.PullRequest,
-    GitHub.Webhook.EventSupport,
     Repo
   }
 
   @doc """
-  Handles a GitHub event based on its type.
+  Handles a fully supported GitHub event based on its type and action.
+
+  The handling process consistes of 3 steps
+
+  - create event record marked as "unprocessed"
+  - mark event record as processing and handle it
+  - mark event record as processed or errored depending on handling outcome
   """
-  def handle(type, id, payload) do
-    with %{} = params <- build_params(type, id, payload),
-         {:ok, %GithubEvent{} = event} <- params |> create_event(),
-         {:ok, %GithubEvent{status: "processing"} = event} <- event |> Event.start_processing
-    do
-      payload |> do_handle(type) |> Event.stop_processing(event)
+  @spec handle_supported(String.t, String.t, map) :: {:ok, GithubEvent.t}
+  def handle_supported(type, id, %{} = payload) do
+    with {:ok, %GithubEvent{} = event} <- type |> build_params(id, "unprocessed", payload) |> create_event() do
+      payload |> apply_handler(type) |> Event.stop_processing(event)
     end
   end
```

Comment:
Removed what's basically an `if` from this module and simply _flattened it out_ from the POV of the controller. Keeps things simpler, allows us to test more directly, and peels off an abstraction layer.